robot_skills/src/robot_skills/hero.py

```diff
@@ -50,6 +50,63 @@ def __init__(self, wait_services=False):
 
         self.configure()
 
+    def move_to_inspect_pose(self, inspect_target):
+        """
+        This poses the robot for an inspect.
+        :param inspection_position: kdl.Frame with the pose of the entity to be inspected.
+        """
+        # calculate the arm_lift_link which must be sent
+        z_over = 0.3  # height the robot should look over the surface
+        z_hh = 0.9  # height of the robots head at z_arm=0
+        torso_to_arm_ratio = 2  # motion of arm/motion of torso
+        z_head = inspect_target.z() + z_over
+
+        # check whether moving the arm is necessary
+        if z_head < 1.2:
+            rospy.loginfo('Entity is low enough. we dont need to move the arm')
+            return True
+
+        # saturate the arm lift goal
+        z_arm = (z_head - z_hh) * torso_to_arm_ratio
+        if z_arm > 0.69:
+            z_arm = 0.69
+            rospy.logwarn('Warning: looking at excessively high surface')
+        elif z_arm < 0.0:
+            z_arm = 0.0
+            rospy.logwarn('Surface is low enough, we dont need to move the arm.')
+
+        arm = self.get_arm()
+
+        # turn the robot
+        rotation = 1.57
+        rotation_speed = 1
+        rotation_duration = rotation / rotation_speed
+        if arm.has_joint_goal('arm_out_of_way'):
```

Comment:
I would check this in the `__init__`:
```python
assert arm.has_joint_goal('arm_out_of_way`), "arm_out_of_way joint goal is not defined in ..."
```
This way, you know on startup that something's missing instead of halfway through your challenge.